cmd/payload-testing-prow-plugin/filetestresolver.go

```diff
@@ -2,49 +2,38 @@ package main
 
 import (
 	"fmt"
-	"path/filepath"
-	"strings"
-
-	"github.com/sirupsen/logrus"
 
 	"github.com/openshift/ci-tools/pkg/api"
-	"github.com/openshift/ci-tools/pkg/config"
 	jc "github.com/openshift/ci-tools/pkg/jobconfig"
+	"github.com/openshift/ci-tools/pkg/load/agents"
 )
 
 type fileTestResolver struct {
-	// TODO: Refresh when the data on files change
-	tuples map[string]api.MetadataWithTest
+	configAgent agents.ConfigAgent
 }
 
-func newFileTestResolver(dir string) (testResolver, error) {
-	ret := &fileTestResolver{
-		tuples: map[string]api.MetadataWithTest{},
-	}
-	if err := config.OperateOnCIOperatorConfigDir(dir, func(configuration *api.ReleaseBuildConfiguration, info *config.Info) error {
-		if !strings.HasPrefix(filepath.Base(info.Filename), "openshift-release-master") {
-			return nil
-		}
-		for _, element := range configuration.Tests {
-			if element.Cron != nil || element.Interval != nil || element.ReleaseController {
-				jobName := configuration.Metadata.JobName(jc.PeriodicPrefix, element.As)
-				ret.tuples[jobName] = api.MetadataWithTest{
-					Metadata: configuration.Metadata,
-					Test:     element.As,
-				}
-				logrus.WithField("jobName", jobName).WithField("test", element.As).Info("loaded job for test")
-			}
-		}
-		return nil
-	}); err != nil {
-		return nil, fmt.Errorf("failed to operate on ci operator config dir %s: %w", dir, err)
-	}
-	return ret, nil
+func newFileTestResolver(configAgent agents.ConfigAgent) (testResolver, error) {
+	return &fileTestResolver{configAgent: configAgent}, nil
```

Comment:
since both `fileTestResolver` and `newFileTestResolver` are unexported there is no reason to have `newFileTestResolver()`